airflow/api_connexion/endpoints/task_instance_endpoint.py

```diff
@@ -251,18 +251,8 @@ def post_clear_task_instances(dag_id: str, session=None):
     task_instances = dag.clear(get_tis=True, **data)
     if not data["dry_run"]:
         clear_task_instances(
-            task_instances,
-            session,
-            dag=dag,
-            activate_dag_runs=False,  # We will set DagRun state later.
+            task_instances, session, dag=dag, dag_run_state=State.RUNNING if reset_dag_runs else None
         )
-        if reset_dag_runs:
-            dag.set_dag_runs_state(
```

Comment:
`dag.set_dag_runs_state()` is no longer used after this PR. I feel we should remove it. But I'm not sure if that's a good thing to do in a bug fix. So I didn't do that here.

Reply:
I’d say it’s better to remove it here; if the code is dead after a bug fix, that code is simply wrong and should not exist 🙂

Reply:
I'd like to remove it, but this is potentially a breaking change 😢 (who knows what users might be doing.) so please add it back and make it issue a DeprecationWarning instead.

Reply:
Done.